pythonforandroid/bootstrap.py

```diff
@@ -138,36 +139,52 @@ def run_distribute(self):
         self.distribution.save_info(self.dist_dir)
 
     @classmethod
-    def list_bootstraps(cls):
+    def all_bootstraps(cls):
         '''Find all the available bootstraps and return them.'''
         forbidden_dirs = ('__pycache__', 'common')
         bootstraps_dir = join(dirname(__file__), 'bootstraps')
+        result = set()
         for name in listdir(bootstraps_dir):
             if name in forbidden_dirs:
                 continue
             filen = join(bootstraps_dir, name)
             if isdir(filen):
-                yield name
+                result.add(name)
+        return result
 
     @classmethod
     def get_bootstrap_from_recipes(cls, recipes, ctx):
```

Comment:
feel like this method is big now and could be split furthermore by concerns, what do you think?

Reply:
sorry I don't see it :joy: the previous one I finally split was due to nesting not to length, I think "it's big" is still a somewhat poor criteria (on its own) for a split. UNLESS you want to be able to test the sub parts separately of course, although I'm not sure that is necessary...? I would personally leave it as is :woman_shrugging:

Reply:
Yes definitely big is a poor and arbitrary criteria, just like the 80 char lines of the pep. It probably started from some feelings and adjusted over time/researches. That "it's big" feeling would vary from one to another of course.
I wanted to share the feeling before looking for solution to see maybe luckily something super clever would have come to your or other people mind :smile:

Reply:
Then see taking a quick look I already see some possible split since the comments and the logs say it all, for instance:
`# Find out which bootstraps are acceptable:`
Then a big block and then:
`info('Found {} acceptable bootstraps: {}`
For me that definitely feels like a split. Not only it will make things more readable:
`acceptable_bootstraps = find_acceptable_bootstraps(...)`
But also make it more unit testable :wink:

Reply:
Yeah and it also obscures the control flow and introduces interdependencies and more interfaces making changes later harder. I definitely see your point but I don't think chasing that far after tests is necessarily good, so if you ask me I think this is better off as it is UNTIL we actually want to use it separately, or it reaches a length where this is more warranted. It is also not such a deeply complex and core functionality that I'd think this hurts us a lot not being tested more granularly.

But as usual, I'm happy to leave the final decision to you guys

Reply:
That's seemed like an easy refactor to me, but OK :cry: 
As usual I'm nitpicking, but won't block a PR for something like this. So if you don't feel like it let's leave it for now. Maybe at some point it bugs so much that I'll do it myself.
Thanks for discussing it :+1:

Reply:
oh no I haven't convinced you even in the slightest have I :scream: oh dear. maybe I'm wrong then, I'll split it up :+1: I mean I can see your side a little, and in the long run I feel like you had so far a slightly better grasp at writing nice code than I have

Reply:
That's so nice of you :blush: 
Thank you @JonasT !